openrtb3/openrtb3.go

```diff
@@ -0,0 +1,162 @@
+	WSeat   *Bool           `json:"wseat,omitempty"`   // Default: 1 (true). Flag that determines the restriction interpretation of the “seat” array, where 0 (false) = block list, 1 (true) = whitelist.
+	CData   string          `json:"cdata,omitempty"`   // Allows bidder to retrieve data set on its behalf in the exchange’s cookie (refer to “cdata” in Object: Response) if supported by the exchange. The string must be in base85 cookie-safe characters.
+	Source  *Source         `json:"source,omitempty"`  // A Source object that provides data about the inventory source and which entity makes the final decision.
+	Item    []Item          `json:"item"`              // Required. Array of Item objects (at least one) that constitute the set of goods being offered for sale.
+	Package Bool            `json:"package,omitempty"` // Flag to indicate if the Exchange can verify that the items offered represent all of the items available in context (e.g., all impressions on a web page, all video spots such as pre/mid/post roll) to support road-blocking, where 0 = no, 1 = yes.
+	Context json.RawMessage `json:"context,omitempty"` // Recommended. Layer-4 domain object structure that provides context for the items being offered conforming to the specification and version referenced in “openrtb.domainspec” and “openrtb.domainver”. For AdCOM v1.x, the objects allowed here all of which are optional are one of the “DistributionChannel” subtypes (i.e., “Site”, “App”, or “Dooh”), “User”, “Device”, “Regs”, and any objects subordinate to these as specified by AdCOM.
+	Ext     json.RawMessage `json:"ext,omitempty"`     // Optional exchange-specific extensions.
+}
+
+// Source object carries data about the source of the transaction including the unique ID of the transaction itself, source authentication information, and the chain of custody.
+type Source struct {
+	TID    string          `json:"tid,omitempty"`    // Recommended. Transaction ID that must be common across all participants throughout the entire supply chain of this transaction. This also applies across all participating exchanges in a header bidding or similar publisher-centric broadcast scenario.
+	DS     string          `json:"ds,omitempty"`     // Recommended. Digital signature used to authenticate this request computed by the publisher or its trusted agent from the transaction digest string "tid:digest", where ‘tid’ matches the "tid" attribute and ‘digest’ is a string composed of an immutable portion of domain objects as defined in the domain specification used for this request. Refer to Section “Inventory Authentication” for more details.
+	DSGVer int8            `json:"dsgver,omitempty"` // Recommended. Format version of the digest string that was digitally signed to produce “ds”. Refer to Section “Inventory Authentication” for more details.
```

Comment:
TODO: check ads.txt doc for allowed list of versions, int8 may not be large enough to hold them all.

Reply:
This attribute seems to be gone from FINAL spec.